### ISSUE TITLE
Configurable Light Alarm

### ISSUE THREAD

**Issue by Natrixz** [theyosh/TerrariumPI#292]
Hi Yosh,

Sorry for bugging you;
Is it possible to generate alarms between certain times?
My use-case is that I have a MiFlora Light sensor in use, in case my Lights don't go on (or suddenly go off)
It apparently happens occasionally with my EG-PM2 and a reboot fixes this. (TerrariumPi will show the lights as "on" when they are off)

Right now I manually inspect the dashboard but I would rather have it generate an alarm or have it auto-restart =)
Or maybe go looking for the cause.....but still would be nice to have this option =)

**Comment by djarrard** [comment]
Is the state of the switch accurately reflected on the website as being on/off when it shouldn’t be?

If so, you could schedule a script to run that checks the state of the switch via the api and notifies you when it’s not working properly. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Natrixz** [comment]
No sadly not ☹
It shows it as “on” and even toggling it will not actually toggle it.
Im suspecting it’s losing the connection and doesn’t restore it

Sent from Mail for Windows 10

**Comment by theyosh** [comment]
Is it a network version? I do get the feeling that it will not always keep the connection open. So I have a ping running to the  EG-PM2 power switch.

But, I think it should be possible to use the light sensors for this. But I do need some more thinking about this, as the current notification system is probably not suitable for this. So that needs also some extension.

**Comment by Natrixz** [comment]
It actually is the USB-version.
Makes it even stranger because it always used to work fine….
It’s not until recently that it’s starting to do this and I just haven’t had the time yet to try and troubleshoot it :(

Might just be an USB-issue, never the less would be nice if we could get notifications from it =)
(might be a nice moment to add more extensions to the notifications)

**Comment by theyosh** [comment]
In order to know if a light is on by a sensor, we have to have some threshold values. This can be entered manually, or can be calculated.

So I am not sure how to tackle this. I was thinking of using 24 hour data, pick the min and max in that period. Divide it by 2 get the average. And when the sensor is above it, the light is on. Below is off.

But then, the software needs at least 24 hours of data running the right schema. Could be a bit confusing to setup.

Manually setting the 'average' value could also be an option. But that will also need some data to figure out the right value.

Any idea's from your side about this?

**Comment by Natrixz** [comment]
Why not just start with entering it manually?
That way you can specify the threshold yourself, above value X, lights are on.
Below value Y (or maybe even value X) lights are off

After that you/we can always progress into using a calculated value right?

**Comment by theyosh** [comment]
Sorry for the late reply, but for now I put it on the wish list...

**Comment by theyosh** [comment]
So after some years, I finally had time to add this feature. And it has become pretty basic, almost the same way you requested.

At a light sensor, you can now add a threshold value which indicates the value when the light is on. Or 'on enough'. 
![image](https://github.com/theyosh/TerrariumPI/assets/8080177/b7b65e78-afbb-40b7-80f7-904977cab4a0)

Enter a value like the average of the on value that is measured. This can be done with every light sensor you have.

You will get a small light bulb indicating that the light threshold is set. And will light up yellow when the light is 'on' according the threshold value.

![image](https://github.com/theyosh/TerrariumPI/assets/8080177/a6856363-a05f-4a89-939f-735262da8e8a)


Next in the light area you can select the light sensors which are then used for checking if the lights are on when needed. You can select light sensors, or LDR buttons, which are also light sensors.

![image](https://github.com/theyosh/TerrariumPI/assets/8080177/8a1bbd62-5088-4c68-87d0-e62b3fdd6580)

When the lights are off based on sensors and should be on, or vise versa, there will be a notification message. 

The notification message can be configured at the notification messages. Use the message type `Lights area wrong state`

![image](https://github.com/theyosh/TerrariumPI/assets/8080177/8e948a79-0720-4726-9970-bc71e2fbfe0a)

And now you can make you message with the placeholders on the left.

When mutliple light sensors are selected, an average is calculated for all the threshold values and form all the current value. The averages are compared.
And when LDR buttons are used, at least half those sensors should be on to be on for all.

So with a combination of light sensors and LDR buttons, you need to be within the averages of the sensors, and at least the half of the LDR buttons needs to be on.

Is this something you where looking for al those years ;)
